src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -1759,6 +1763,24 @@ private Symbol GetDeclaredMember(NamespaceOrTypeSymbol container, TextSpan decla
 
             // look for any member with same declaration location
             var collection = name != null ? container.GetMembers(name) : container.GetMembersUnordered();
+            if (isKnownToBeANamespace)
+            {
+                // Filter the collection to only include namespace symbols. This will not allocate a new instance for
+                // the common case where all symbols in the collection are already namespace symbols.
+                var namespaces = collection.WhereAsArray(symbol => symbol is NamespaceSymbol);
```

Comment:
Consider avoiding this allocation

Reply:
➡️ This already avoids allocations in the common case. The remaining allocation occurs in an edge case that is also a compile-time error, so users will be "strongly encouraged" to avoid it. 😄